core/services/workflows/monitoring.go

```diff
@@ -30,12 +30,16 @@ type engineMetrics struct {
 	workflowExecutionLatencyGauge            metric.Int64Gauge // ms
 	workflowStepErrorCounter                 metric.Int64Counter
 	workflowInitializationCounter            metric.Int64Counter
-	engineHeartbeatCounter                   metric.Int64Counter
-	workflowCompletedDurationSeconds         metric.Int64Histogram
-	workflowEarlyExitDurationSeconds         metric.Int64Histogram
-	workflowErrorDurationSeconds             metric.Int64Histogram
-	workflowTimeoutDurationSeconds           metric.Int64Histogram
-	workflowStepDurationSeconds              metric.Int64Histogram
+
+	// Deprecated: use the gauge instead
+	engineHeartbeatCounter metric.Int64Counter
+	engineHeartbeatGauge   metric.Int64Gauge
```

Comment:
why a gauge versus a counter to count ticks?

Reply:
The gauge allows us to do sum by / count by aggregations and would enable us to create dashboards tracking how many nodes are running a workflow for instance